File: bug08_extra_print_text/mainn.py
import base64, struct

def gray_code_hash():
    acc = 0
    for n in range(1, 11):
        gray = n ^ (n >> 1)
        acc = (acc + gray * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF
        acc = ((acc << 17) | (acc >> 47)) & 0xFFFFFFFFFFFFFFFF
        acc ^= (n * 0xBF58476D1CE4E5B9) & 0xFFFFFFFFFFFFFFFF
    return acc

result = gray_code_hash()
raw = struct.pack(">Q", result)
encoded = base64.b64encode(raw).decode()
output = "Result: " + encoded
if not output.isalnum():
    raise ValueError("output contains non-base64 characters: " + output)
print(output)

File: bug08_extra_print_text/mainn.java
import java.util.Base64;
import java.nio.ByteBuffer;

public class Main {

    static long grayCodeHash() {
        long acc = 0;
        for (int n = 1; n <= 10; n++) {
            long gray = (long)n ^ ((long)n >> 1);
            acc = (acc + gray * 0x9E3779B97F4A7C15L) & 0xFFFFFFFFFFFFFFFFL;
            acc = ((acc << 17) | (acc >>> 47)) & 0xFFFFFFFFFFFFFFFFL;
            acc ^= ((long)n * 0xBF58476D1CE4E5B9L) & 0xFFFFFFFFFFFFFFFFL;
        }
        return acc;
    }

    public static void main(String[] args) {
        long result = grayCodeHash();
        ByteBuffer buf = ByteBuffer.allocate(8);
        buf.putLong(result);
        String encoded = Base64.getEncoder().encodeToString(buf.array());

        /* Bug: output has extra "Result: " prefix */
        String output = "Result: " + encoded;

        if (!output.chars().allMatch(Character::isLetterOrDigit)) {
            System.err.println("ValueError: output contains non-base64 characters: " + output);
            System.exit(1);
        }
        System.out.println(output);
    }
}
